lib/pluginLoader.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+module.exports = PluginLoader;
+
+function PluginLoader (Nocca) {
+
+    var self = this;
+
+    var loadedPlugins = {};
+    var instantiatedPlugins = {};
+
+    self.registerPlugins = registerPlugins;
+    self.instantiatePlugin = instantiatePlugin;
+
+    function registerPlugins (plugins) {
+
+        plugins.forEach(function (plugin) {
+
+            if (typeof plugin === 'string') {
+                // name for plugin!
+
+                // load from the plugins dir, should return an object
+                plugin = require(__dirname + '/plugins/' + plugin);
+
+            }
+
+
+            // if plugin was an object, or was require()'d as object, proceed
+            if (typeof plugin === 'object') {
+                // plugin definition!
+
+                registerPlugin(plugin);
+
+            }
+
+        });
+
+    }
+
+    function registerPlugin (pluginDefinition) {
+
+        // make sure it's valid before continuing
+        validatePluginDefinition(pluginDefinition);
+
+        // confirm plugin is not yet loaded
+        if (typeof loadedPlugins[pluginDefinition.id] === 'undefined') {
```

Comment:
Fixed it, comparing constructor function when id's match, then error when different.